cmd/query/app/otel/otlp_translator.go

```diff
@@ -0,0 +1,353 @@
+}
+
+func jLogsToOTLP(logs []model.Log) []*v1.Span_Event {
+	events := make([]*v1.Span_Event, len(logs))
+	for i, l := range logs {
+
+		var name string
+		var ignoreKeys map[string]bool
+		if messageTag, ok := model.KeyValues(l.GetFields()).FindByKey(tracetranslator.TagMessage); ok {
+			name = messageTag.GetVStr()
+			ignoreKeys = map[string]bool{}
+			ignoreKeys[tracetranslator.TagMessage] = true
+		}
+
+		events[i] = &v1.Span_Event{
+			TimeUnixNano: uint64(l.GetTimestamp().UnixNano()),
+			Name:         name,
+			Attributes:   jTagsToOTLP(l.GetFields(), ignoreKeys),
+		}
+	}
+	return events
+}
+
+func jReferencesToOTLP(refs []model.SpanRef, excludeParentID model.SpanID) []*v1.Span_Link {
+	if len(refs) == 0 || len(refs) == 1 && refs[0].SpanID == excludeParentID && refs[0].RefType == model.ChildOf {
+		return nil
+	}
+	var links []*v1.Span_Link
+	for _, r := range refs {
+		if r.SpanID == excludeParentID && r.GetRefType() == model.ChildOf {
+			continue
+		}
+		links = append(links, &v1.Span_Link{
+			TraceId: uint64ToTraceID(r.TraceID.High, r.TraceID.Low),
+			SpanId:  uint64ToSpanID(uint64(r.SpanID)),
+		})
+	}
+
+	return links
+}
+
+func getSpanStatus(tags []model.KeyValue) (*v1.Status, map[string]bool) {
+	statusCode := v1.Status_STATUS_CODE_UNSET
+	statusMessage := ""
+	statusExists := false
+
+	ignoreKeys := map[string]bool{}
+	kvs := model.KeyValues(tags)
+	if _, ok := kvs.FindByKey(tracetranslator.TagError); ok {
+		statusCode = v1.Status_STATUS_CODE_ERROR
+		statusExists = true
+		ignoreKeys[tracetranslator.TagError] = true
+	}
+	if tag, ok := kvs.FindByKey(tracetranslator.TagStatusCode); ok {
+		statusExists = true
+		if code, err := getStatusCodeValFromTag(tag); err == nil {
+			statusCode = v1.Status_StatusCode(code)
+			ignoreKeys[tracetranslator.TagStatusCode] = true
+		}
+		if tag, ok := kvs.FindByKey(tracetranslator.TagStatusMsg); ok {
+			statusMessage = tag.GetVStr()
+			ignoreKeys[tracetranslator.TagStatusMsg] = true
+		}
+	} else if tag, ok := kvs.FindByKey(conventions.AttributeHTTPStatusCode); ok {
+		statusExists = true
+		if code, err := getStatusCodeFromHTTPStatusTag(tag); err == nil {
+			// Do not set status code in case it was set to Unset.
+			if v1.Status_StatusCode(code) != v1.Status_STATUS_CODE_UNSET {
+				statusCode = v1.Status_StatusCode(code)
+			}
+
+			if tag, ok := kvs.FindByKey(tracetranslator.TagHTTPStatusMsg); ok {
+				statusMessage = tag.GetVStr()
+			}
+		}
+	}
+
+	if statusExists {
+		return &v1.Status{
+			Code:    statusCode,
+			Message: statusMessage,
+		}, ignoreKeys
+	}
+	return nil, nil
+}
+
+func getStatusCodeValFromTag(tag model.KeyValue) (int, error) {
+	var codeVal int64
+	switch tag.GetVType() {
+	case model.ValueType_INT64:
+		codeVal = tag.GetVInt64()
+	case model.ValueType_STRING:
+		i, err := strconv.Atoi(tag.GetVStr())
+		if err != nil {
+			return 0, err
+		}
+		codeVal = int64(i)
+	default:
+		return 0, fmt.Errorf("invalid status code attribute type: %s", tag.GetKey())
```

Comment:
```suggestion
		return 0, fmt.Errorf("invalid status code attribute type: %q", tag.GetKey())
```

Reply:
key is string

Reply:
Right, and %q will make the string be wrapped in quotes.